content/docs/04-ai-sdk-ui/03-chatbot-message-persistence.mdx

```diff
@@ -325,3 +325,171 @@ When the client reloads the page after a disconnect, the chat will be restored f
   the case where the client reloads the page after a disconnection, but the
   streaming is not yet complete.
 </Note>
+
+## Resuming ongoing streams
+
+<Note>This feature is experimental and may change in future versions.</Note>
+
+The `useChat` hook has experimental support for resuming an ongoing chat generation stream by any client, either after a network disconnect or by reloading the chat page. This can be useful for building applications that involve long-running conversations or for ensuring that messages are not lost in case of network failures.
+
+The following are the pre-requisities for your chat application to support resumable streams:
+
+- Installing the [`resumable-stream`](https://www.npmjs.com/package/resumable-stream) package that helps create and manage the publisher/subscriber mechanism of the streams.
+- Creating a [Redis](https://vercel.com/marketplace/redis) instance to store the stream state.
+- Creating a table that tracks the stream IDs associated with a chat.
+
+To resume a chat stream, you will use the `experimental_resume` function returned by the `useChat` hook. You will call this function during the initial mount of the hook inside the main chat component.
+
+```filename="app/components/chat.tsx"
```

Comment:
ts

Reply:
also will this break when docs are deplyed?

Reply:
hmm not sure, but I updated!